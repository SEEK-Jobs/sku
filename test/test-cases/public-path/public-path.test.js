const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const waitForUrls = require('../../utils/waitForUrls');
const getAppSnapshot = require('../../utils/getAppSnapshot');
const startAssetServer = require('../../utils/assetServer');
const targetDirectory = `${__dirname}/dist`;

describe('public path', () => {
  describe('build', () => {
    let closeAssetServer;

    beforeAll(async () => {
      await runSkuScriptInDir('build', __dirname);
      closeAssetServer = startAssetServer(4001, targetDirectory);
      await waitForUrls('http://localhost:4001');
    });

    afterAll(() => {
      closeAssetServer();
    });

    it('should create valid app with no unresolved resources', async () => {
      const app = await getAppSnapshot('http://localhost:4001/static');
      expect(app).toMatchSnapshot();
    });
  });
});
