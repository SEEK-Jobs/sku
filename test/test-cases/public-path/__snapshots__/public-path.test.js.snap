// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`public path build should create valid app with no unresolved resources 1`] = `
SOURCE HTML: 
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="/static/main-d72d68abb80c2eed0357.css"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="/static/runtime-ae11dd820f2b1b4ccf66.js"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="/static/vendors~main-fabca146acc8059ae7df.js"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="/static/main-f52d86b1762b403f4a0e.js"
    >
  </head>
  <body>
    <div id="app">
      <div class="_1oKZ42c">
      </div>
    </div>
    <script>
      window.__LOADABLE_REQUIRED_CHUNKS__ = [];
    </script>
    <script async
            data-chunk="main"
            src="/static/runtime-ae11dd820f2b1b4ccf66.js"
    >
    </script>
    <script async
            data-chunk="main"
            src="/static/vendors~main-fabca146acc8059ae7df.js"
    >
    </script>
    <script async
            data-chunk="main"
            src="/static/main-f52d86b1762b403f4a0e.js"
    >
    </script>
  </body>
</html>

POST HYDRATE DIFF: No diff
WARNINGS: Array []
ERRORS: Array []
`;
